src/Bounds.cpp

```diff
@@ -402,14 +402,18 @@ class Bounds : public IRVisitor {
             // Assume no overflow for float, int32, and int64
             if (!op->type.is_float() && (!op->type.is_int() || op->type.bits() < 32)) {
                 if (interval.has_upper_bound()) {
-                    Expr no_overflow = (cast<int>(a.max) + cast<int>(b.max) == cast<int>(interval.max));
+                    // TODO(rootjalex): Can't catch overflow of UInt(64) currently.
+                    Type t = op->type.is_uint() ? UInt(64) : Int(32);
```

Comment:
Why do we use 64 bits for the unsigned case but only 32 for signed?

Reply:
Overflow has defined behavior for all unsigned types, but only for signed integers with bitwidths `<= 16`. Technically the upcasted type for unsigned integers should be `UInt(128)` so we can catch overflow for `UInt(64)` but Halide is not capable of such bitwidths at the moment

Reply:
OK, that makes sense, but what is the eventual TODO action here? (i.e., it implies it's something we can and should fix; what does that fix look like, does it need a tracking issue, etc?)

Reply:
There are some ways to catch overflow for the highest bitwidth, despite not being capable of representing that type. The TODO is to remind me to test out some of them - but perhaps it's not reasonable, as I expect (and perhaps @abadams can confirm/deny) that overflow with `UInt(64)` is uncommon enough that bounds inference can likely not worry about this case.

I can open an issue to track this, if deemed worth it!

Reply:
Added in 87383fb7c